Dockerfile

```diff
@@ -1,164 +1,90 @@
-USER root
-RUN apk del .build-deps \
-    && apk add bash openjdk11
+# copy files 
+WORKDIR /tmp
+COPY . .
 
-RUN java -version
-RUN javac -version
+# copy all package.json files
+RUN find . -maxdepth 3 -mindepth 2 -type f -name "package.json" ! -path '**node_modules**' ! -path './docker' -exec cp --parents '{}' $DEST_FOLDER \;
+COPY package.json $DEST_FOLDER
+COPY yarn.lock $DEST_FOLDER
 
-# make sure of cache folder perms
-RUN chown -R node:node /home/unlock/yarn-cache
+# copy al yarn related files
+COPY .yarnrc.yml $DEST_FOLDER
+COPY .prettierrc $DEST_FOLDER
+COPY .yarn/plugins $DEST_FOLDER/.yarn/plugins
+COPY .yarn/releases $DEST_FOLDER/.yarn/releases
+RUN chown -R node:node $DEST_FOLDER
 
-USER node
+# specify yarn cache folder location (to be used by docker buildkit)
+RUN echo "cacheFolder: ${DEST_FOLDER}/yarn-cache" >> $DEST_FOLDER/.yarnrc.yml
+RUN mkdir $DEST_FOLDER/yarn-cache
+RUN chown -R node:node $DEST_FOLDER/yarn-cache
 
-# build all packages in packages/**
-RUN yarn build
-
-# copy scripts
-RUN mkdir /home/unlock/scripts
-COPY --chown=node scripts /home/unlock/scripts
-
-# copy app code
-COPY --chown=node ${BUILD_DIR}/ /home/unlock/${BUILD_DIR}/.
+# install deps
+WORKDIR ${DEST_FOLDER}
+RUN yarn
 
-###
-## Need to run subgraph test 
-###
-FROM dev as subgraph 
+#
+# 2. build packages and prepare image for testing/dev
+#
+FROM deps as dev
 
+# enforce perms
 USER root
-RUN apk add postgresql \ 
-    build-base \ 
-    libpq-dev 
-
-ENV GLIBC_REPO=https://github.com/sgerrand/alpine-pkg-glibc
-ENV GLIBC_VERSION=2.30-r0
-RUN set -ex && \
-    apk --update add libstdc++ curl ca-certificates && \
-    for pkg in glibc-${GLIBC_VERSION} glibc-bin-${GLIBC_VERSION}; \
-        do curl -sSL ${GLIBC_REPO}/releases/download/${GLIBC_VERSION}/${pkg}.apk -o /tmp/${pkg}.apk; done && \
-    apk add --allow-untrusted /tmp/*.apk && \
-    rm -v /tmp/*.apk && \
-    /usr/glibc-compat/sbin/ldconfig /lib /usr/glibc-compat/lib
-
-##
-## 3. build the app
-##
-FROM dev as build
+WORKDIR /home/unlock
+RUN chown -R node:node /home/unlock
 
-ARG BUILD_DIR
-ARG PORT
+# default user
+USER node
+COPY --chown=node . .
 
-# additional build step (nb: strip "packages/" to get worspace name)
-RUN yarn workspace @unlock-protocol/${BUILD_DIR/packages\/} build
+# make sure java is installed properly
+RUN java -version
+RUN javac -version
 
-# package everything for prod
-RUN cd $BUILD_DIR && yarn prod-install --pack /home/node/app
+# build all packages in packages/**
+RUN yarn build
 
 ##
-## 4. export a minimal image w only the prod app
+## 3. export image for prod app
 ##
-FROM node:16-alpine as prod
+FROM dev as prod
 
-ARG BUILD_DIR
-ARG PORT
+# default values
 ARG COMMAND="yarn prod"
+ENV BUILD_DIR='locksmith'
 ENV COMMAND=${COMMAND}
 
-USER root
-RUN mkdir /app
-RUN chown node:node /app
-
-WORKDIR /app
-
-# copy package info
-COPY --from=build --chown=node /home/node/app .
+WORKDIR /home/unlock/${BUILD_DIR}
 
 # start command
-EXPOSE $PORT
+EXPOSE 3000
```

Comment:
I know you had questions about this but are we sure we want to hardcode the port in the docker file?
Would be better in the dockercompose config maybe?